cmd/agentless-scanner/awsmain.go

```diff
@@ -0,0 +1,736 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/DataDog/datadog-agent/cmd/agentless-scanner/awsutils"
+	"github.com/DataDog/datadog-agent/cmd/agentless-scanner/devices"
+	"github.com/DataDog/datadog-agent/cmd/agentless-scanner/runner"
+	"github.com/DataDog/datadog-agent/cmd/agentless-scanner/types"
```

Comment:
We usually try to have only the CLI in `cmd` folder and the core logic for the product in `comp` if it has a state or `pkg` if it's pure helpers functions.
Could we split the agentless to follow this pattern ?

Reply:
You seems to have internal state which point toward `comp` folder instead. We aim for the `pkg` folder to be stateless at some point.

Reply:
Which internal state are you referring to?